the-collab-lab/tcl-11-smart-shopping-list
# Feature (Issue #11):  Delete shopping list item(s)

## Description
This Pull Request (PR) adds a new feature to the `<Item />` allowing users to remove items from their shopping list. When a user attempts to remove an item, they are prompted with a dialog screen to confirm that they want to delete the item to prevent accidental deletions. Upon clicking 'YES', all of the associated record(s) in the database to are deleted.

## Related Issue

Closes Issue #11

## Acceptance Criteria


## Type of Changes


|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

![image](https://user-images.githubusercontent.com/32943879/93688366-4a2fbb00-fa93-11ea-8324-30d6669f22eb.png)

### After
![DeleteItems](https://user-images.githubusercontent.com/32943879/93688720-d8a53c00-fa95-11ea-8c1f-997da0167753.gif)

## Testing Steps / QA Criteria
 
- From your terminal, run `git checkout sd-kl-delete-list-item` in the `tcl-11-smart-shopping-list ` repository.
- Once on the `sd-kl-delete-list-item` branch, run npm start to load the server and start the browser.
- Visit the `<Home/>` component at http://localhost:3000.
- Join an existing shopping list or to create a new shopping list, click the 'Create a New Shopping List' button to be redirected to the `<List />` component view. 
- Click the 'Add Item' button' to be redirected to the `<AddItem />` component view and add any item to the shopping list.
- After the page reloads and the list of items is displayed, click the 'trash can' icon to remove the item from the shopping list view and Firebase database.
